lnrpc/routerrpc/router.proto

```diff
@@ -125,8 +124,27 @@ message PaymentStatus {
     The taken route when state is SUCCEEDED.
     */
     lnrpc.Route route = 3;
+
+    /**
+    List of all shard statuses to this payment_hash [EXPERIMENTAL].
+    */
+    repeated ShardStatus shards = 4 [json_name = "shards"];
 }
 
+message ShardStatus {
+    /// Current state the shard is in.
+    PaymentState state = 1;
+
+    /**
+    The preimage of the shard when state is SUCCEEDED.
+    */
+    bytes preimage = 2;
```

Comment:
For mpp, this is a payment level property. Do we report it here as a prep for amp? Maybe it is better to create a payment-level property 'payment proof' and only populate it for mpp payments.